Add dialog to enter status text

mightymop opened the pull request.

sualko commented on src/ui/Roster.ts:
```diff
@@ -359,19 +360,26 @@ export default class Roster {
-            if (typeof onUserRequestsToGoOnline === 'function') {
-               onUserRequestsToGoOnline();
-            }
+         else
```
Returning inside the if clause would help to reduce indention and enhance the diff too.

mightymop replied:
done

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,68 @@
+{
+   let account = Client.getAccountManager().getAccounts()[0];
+   let statustext = dialog.getDom().find('#extended_status_text').val().toString();
+   let item = $build('status',{xmlns:'http://jabber.org/protocol/status'}).c('text').t(statustext).tree();
```
We should move this to a service.

mightymop replied:
ok i have moved the build code for the packets to pep service

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,68 @@
+function onSubmit()
+{
+   let account = Client.getAccountManager().getAccounts()[0];
+   let statustext = dialog.getDom().find('#extended_status_text').val().toString();
```
Use e.g. form name instead of id.

mightymop replied:
ok i have moved the build code for the packets to pep service

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,68 @@
+import Dialog from '../Dialog'
+import Client from '../../Client'
+//import JID from '../../JID'
```
Should be removed.

mightymop replied:
removed

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,68 @@
+
+   account.getContact().setStatus(statustext);
+
+   account.getConnection().getPEPService().publish('http://jabber.org/protocol/status',item,'current').then(function(result) {
```
In which XEP is this defined?

mightymop replied:
oh i now see it is not official... so we could remove it or not... up to you...
found it here: http://www.funkroom.net/xep/xep-status.html

sualko commented on template/roster.hbs:
```diff
@@ -5,29 +5,33 @@
-            <li data-presence="dnd" class="jsxc-dnd">{{t "dnd"}}</li>
-            <li data-presence="offline" class="jsxc-offline">{{t "Offline"}}</li>
-      {{/menu}}
+   <div class="jsxc-real-bottom">
```
Real bottom was probably not intentionally included in this pr.

mightymop replied:
right... C&P error ;D

sualko commented on src/connection/services/PEP.ts:
```diff
@@ -23,6 +23,16 @@ export default class PEP extends AbstractService {
       }
    }
 
+   public publisStatus(statustext:string)
```
Typo

mightymop replied:
removed...

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,66 @@
+function sendStatus()
+{
+
+   let account = Client.getAccountManager().getAccounts()[0];
```
We should loop through all accounts

mightymop replied:
added account choice

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,66 @@
+   dialog = new Dialog(content);
+   let dom = dialog.open();
+
+   $(dom).ready(function(){
```
No need to wait

mightymop replied:
changed...

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,66 @@
+
+function getStatus()
+{
+     let account = Client.getAccountManager().getAccounts()[0];
```
We should check all accounts and than take the first one.

mightymop replied:
added account choice

sualko commented on src/ui/dialogs/exstatus.ts:
```diff
@@ -0,0 +1,66 @@
+function removeStatus(ev) {
+   ev.preventDefault();
+
+   let account = Client.getAccountManager().getAccounts()[0];
```
Loop

mightymop replied:
added account choice

sualko reviewed (changes requested):
In my opinion we should not implement http://www.funkroom.net/xep/xep-status.html, because it's lacking crucial information. The biggest issue is, that it's not configuring a single item node. So we would probably end up with multiple old status messages. No big deal to fix it (see omemo), but I don't see a need for it either.

Otherwise looks good.

mightymop commented:
> 
> 
> In my opinion we should not implement http://www.funkroom.net/xep/xep-status.html, because it's lacking crucial information. The biggest issue is, that it's not configuring a single item node. So we would probably end up with multiple old status messages. No big deal to fix it (see omemo), but I don't see a need for it either.
> 
> Otherwise looks good.

i removed that unofficial xep..

sualko commented:
I merged it manually, because I changed some things:

- I removed the account and presence selection, because this would result in confusion. The roster bottom can only show one presence, so it would not be clear which account is using which presence.
- Status can now be added to every presence.
- The status is shown in the bottom part of the roster.

![image](https://user-images.githubusercontent.com/2974196/104520915-3557d400-55fc-11eb-8c17-ed02796032c0.png)

mightymop commented:
> 
> 
> I merged it manually, because I changed some things:
> 
>     * I removed the account and presence selection, because this would result in confusion. The roster bottom can only show one presence, so it would not be clear which account is using which presence.
> 
>     * Status can now be added to every presence.
> 
>     * The status is shown in the bottom part of the roster.
> 
> 
> ![image](https://user-images.githubusercontent.com/2974196/104520915-3557d400-55fc-11eb-8c17-ed02796032c0.png)

very nice improvements...
